### ISSUE TITLE
Accommodate different time points

### ISSUE THREAD

**Issue by jcohenadad** [ivadomed/ivadomed#788]
The [BIDS](https://bids.neuroimaging.io/) way to accommodate time points is to introduce the "ses-X" sub-folder. Example:

~~~
dataset
├── dataset_description.json
├── participants.json
├── participants.tsv
├── sub-ubc01
├── sub-ubc02
├── sub-ubc03
├── sub-ubc04
├── sub-ubc05
├── sub-ubc06
│   ├── ses-01
│   └── ses-02
|       └── anat
|           ├── sub-ubc06_ses-02_T1w.json
|           ├── sub-ubc06_ses-02_T1w.nii.gz
|           ├── sub-ubc06_ses-02_T2w.json
|           ├── sub-ubc06_ses-02_T2w.nii.gz
|           ├── sub-ubc06_ses-02_acq-ax_T2w.json
|           └── sub-ubc06_ses-02_acq-ax_T2w.nii.gz
|
└── derivatives
    └── labels
        └── sub-ubc06
                ├── ses-01
                └── ses-02
                    └── anat
                        ├── sub-ubc06_ses-02_T2w_seg-manual.json
                        ├── sub-ubc06_ses-02_T2w_seg-manual.nii.gz  <------------- manually-corrected spinal cord segmentation
                        ├── sub-ubc06_ses-02_T2w_lesion-manual.json
                        └── sub-ubc06_ses-02_T2w_lesion-manual.nii.gz  <---------- manually-created lesion segmentation
~~~

Does **ivadomed** support this?

**Comment by mariehbourget** [comment]
> The BIDS way to accommodate time points is to introduce the "ses-X" sub-folder. 
Does **ivadomed** support this?

Yes, the new loader using `pybids` fully supports the nested `sessions` folders, the `ses-` entity in filenames and the` _sessions.tsv` when present. All files under `sessions` folders are indexed, and the `ses-label `and `sessions.tsv` metadata is present in the dataframe. It is also tested in our unit tests as I included sessions in the microscopy testing dataset.

**Comment by mariehbourget** [comment]
Regarding multichannel: this is currently only supported in ivadomed for multichannel with different contrasts.

Doing multichannel with different timepoints could be achieved by any of the 3 options Andreanne listed above.
Note: the "partial" hack may not work with dash (`FLAIR-t1`) but should work without (`FLAIRt1`).

**Comment by jcohenadad** [comment]
Thank you @mariehbourget, do you agree that going the "right" way is the right thing to do? copy/paste from @andreanne-lemay comment:

> The loader in its current version will load the session files. The problem comes when we want to aggregate the inputs to have a multichannel model. It will break [here](https://github.com/ivadomed/ivadomed/blob/ccd82d22d31ffa53a2d6455ce3a94b97564d624a/ivadomed/loader/loader.py#L1118) and [here](https://github.com/ivadomed/ivadomed/blob/ccd82d22d31ffa53a2d6455ce3a94b97564d624a/ivadomed/loader/loader.py#L1200). We would need to add a feature in ivadomed to be able to do multichannel not only with modalities (and with session for instance) which is not trivial, but would be very useful in the long term. It's definitely something we should implement at some point.

@ssinad is assigned to any development needed to make it work.

**Comment by mariehbourget** [comment]
> Thank you @mariehbourget, do you agree that going the "right" way is the right thing to do?

Yes, note that the 2 places I flagged as problematic in the "right" way are part of the loader. There may be others places in the code where multichannel is used without the loader part that could need refactoring. From the top of my head, I remember multichannel is present also in the `run_segment_command` in `main.py`.

**Comment by ssinad** [comment]
Do we have tests for this?

**Comment by jcohenadad** [comment]
> Do we have tests for this?

not sure-- but can you please look into this? this is quite urgent-- thx
